relay-server/src/metrics_extraction/generic.rs

```diff
@@ -56,13 +56,20 @@ where
             continue;
         };
 
+        // For extracted metrics we assume the `received_at` timestamp is equivalent to the time
+        // in which the metric is extracted.
+        #[cfg(not(test))]
+        let received_at = UnixTimestamp::now();
```

Comment:
We should do this outside of the loop at the beginning of the function.

Reply:
Good catch!